### PR TITLE
Remove redundent check_function_args

### PR DESCRIPTION
**Pull request by sorpaas**
In `FuncInstance::invoke`, `check_function_args` is immediately called again.

### CODE REVIEW

File: src/module.rs [hunk 1]
```diff
@@ -623,7 +623,6 @@ impl ModuleInstance {
 			}
 		};
 
-		check_function_args(func_instance.signature(), &args)?;
```
Comment (pepyakin) [inline src/module.rs]: In this case, there is not a lot of sense in actually formatting error message inside of `check_function_args`. Maybe we can get rid of it?

**Comment by pepyakin** [comment]
As potential follow-up: Why do we use collect there? Does this involve allocation of a temporary vector?

**Comment by sorpaas** [comment]
@pepyakin Yeah I think that's unnecessary. Removed!

**Comment by Vurich** [comment]
You can still use collect and collect into a `Result<(), _>` instead of a `Result<Vec<_>, _>`, although in this case there's no need to. `()` implements `FromIterator<Item = ()>`.

**Pull request merged**